include/nbl/builtin/hlsl/sort/counting.hlsl

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/builtin/hlsl/sort/common.hlsl"
+#include "nbl/builtin/hlsl/workgroup/arithmetic.hlsl"
+
+#ifndef _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SORT_COUNTING_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace sort
+{
+
+NBL_CONSTEXPR uint32_t BucketsPerThread = ceil((float) BucketCount / WorkgroupSize);
```

Comment:
1. don't ever use floats to do integer math
2. we cannot depend on an outside definition of `WorkgroupSize` or `BucketCount`, everything should be worked out inside `counting`

P.S. after you address the fact you can't declare global smem arrays, accessors, or varialbes, you'll find that `BucketsPerThread` doesn't even need to be constexpr and can be worked out from the counting sort parameters

Reply:
Done